core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -812,6 +827,15 @@ public int numberOfReplicas() {
             return settings.getAsInt(SETTING_NUMBER_OF_REPLICAS, -1);
         }
 
+        public Builder routingPartitionSize(int routingPartitionSize) {
+            settings = Settings.builder().put(settings).put(SETTING_ROUTING_PARTITION_SIZE, routingPartitionSize).build();
+            return this;
+        }
+
+        public int routingPartitionSize() {
+            return settings.getAsInt(SETTING_ROUTING_PARTITION_SIZE, -1);
```

Comment:
fair enough, can you document this behavior? if you want that can be a second PR where we document this across the class?